src/controllers/v2/swaps/index.ts

```diff
@@ -0,0 +1,95 @@
+import { eq, and, inArray } from 'drizzle-orm';
+import { Router } from 'express';
+import Joi from 'joi';
+
+import { DEFAULT_CACHE_TTL } from 'config/constants';
+import { db } from 'database/client';
+import { tokens } from 'database/schema';
+import { swapsTableV2 } from 'database/schema/swaps_v2';
+import { maybeCacheResponse } from 'utils/cache';
+import { toPaginatedResponse } from 'utils/http-response';
+import { createApiQuery, OrderBy, validatePaginatedRequest } from 'utils/pagination';
+import { asyncRoute } from 'utils/route-wrapper';
+import { validate } from 'utils/validation';
+
+const router = Router();
+
+const swapHistoryQuerySchema = Joi.object({
+  user: Joi.string().required(),
+});
+
+router.get(
+  '/',
+  asyncRoute(async (req, res) => {
+    const { user, chainId } = validate(swapHistoryQuerySchema, req.query);
```

Comment:
also, add allowUnknown, otherwise it may fail if `limit` and `cursor` is passed as query
```suggestion
    const { user } = validate(swapHistoryQuerySchema, req.query,  { allowUnknown: true });
```